OpenRA.Mods.Common/Traits/World/ResourceRenderer.cs

```diff
@@ -163,10 +176,16 @@ protected virtual void UpdateRenderedSprite(CPos cell, RendererCellContents cont
 				var maxDensity = type.Info.MaxDensity;
 				var frame = int2.Lerp(0, sprites.Length - 1, density, maxDensity);
 
-				UpdateSpriteLayers(cell, sprites, frame, type.Palette);
+				UpdateSpriteLayers(cell, sprites.GetSprite(frame), type.Palette, sprites.IgnoreWorldTint);
+
+				if (sprites.ShadowStart > 0)
```

Comment:
Wouldn't it make more sense to do this inside `UpdateSpriteLayers`?

Reply:
I rewrote this to be less invasive.